mypy/newsemanal/semanal_main.py

```diff
@@ -230,8 +219,26 @@ def check_type_arguments(graph: 'Graph', scc: List[str], errors: Errors) -> None
                 state.tree.accept(analyzer)
 
 
-def process_abstract_status(graph: 'Graph', scc: List[str], errors: Errors) -> None:
+def calculate_class_properties(graph: 'Graph', scc: List[str], errors: Errors) -> None:
     for module in scc:
         tree = graph[module].tree
         assert tree
-        calculate_abstract_status(tree, errors)
+        for _, node, _ in tree.local_definitions():
+            if isinstance(node.node, TypeInfo):
+                calculate_class_abstract_status(node.node, tree.is_stub, errors)
+                calculate_class_vars(node.node)
+
+
+def calculate_class_vars(info: TypeInfo) -> None:
```

Comment:
Should this leave in `semanal_main`? Maybe move it to `semanal_abstract` and rename the latter to `semanal_classes` (or `semanal_util`)?